### PR TITLE
test(middleware): add unit tests for mothership middleware layer

### PR DESCRIPTION
**Pull request by irfankabir02**
## Summary

Adds unit tests for the mothership middleware layer, which previously had near-zero coverage across its 19 components. Tests use FastAPI `TestClient` for integration paths and direct invocation for pure-logic helpers, with `autouse` fixtures isolating global singletons (`MERIT_SCORING_ENGINE`, the circuit-breaker manager).

## Files added

| File | Coverage target |
|------|----------------|
| `tests/unit/test_admission_dispatch.py` | `AdmissionGateMiddleware.dispatch()` — banner, budget, origin, body, merit gates |
| `tests/unit/test_circuit_breaker.py` | `Circuit`, `CircuitBreakerManager`, `CircuitBreakerMiddleware` |
| `tests/unit/test_security_enforcer.py` | `SecurityEnforcerMiddleware`, `EnforcerMetrics`, `SecurityViolation` |
| `tests/unit/test_security_headers.py` | Standalone `security_headers` module + `__init__` variant |
| `tests/unit/test_rate_limit.py` | `RateLimitMiddleware` (in-memory) + `CognitiveRedisRateLimitMiddleware` (fallback) |
| `tests/unit/test_request_size.py` | `RequestSizeLimitMiddleware` boundary conditions |
| `tests/unit/test_versioning_middleware.py` | `VersioningMiddleware` + `VersionMetadata` header injection / registry |
| `tests/unit/test_drt_middleware.py` | `ComprehensiveDRTMiddleware` signatures, normalization, similarity, escalation, dispatch |
| `tests/unit/test_usage_tracking.py` | `UsageTrackingMiddleware` gating + `UsageMeter` cost units |

## Test plan

- [ ] CI unit stage passes: `uv run pytest tests/unit/ -v --tb=short`
- [ ] No ruff errors beyond accepted ANN201/ANN202 (same pattern as the existing test suite)
- [ ] All tests tagged `@pytest.mark.unit`

## Notes for reviewer

- These tests cannot be executed in the authoring sandbox (no network to install `fastapi`/`starlette`), so they are validated by static reasoning + ruff here and rely on CI to run.
- ⚠️ **The repo's `Lint` CI job is failing on pre-existing debt** (8085 `ANN001/201/202/204` errors in `workspace/mcp/servers/`, unrelated to this PR; `main` has been red since at least June 1). Because `Test` depends on `Lint`, the `Test` stage is currently **skipped**, so these new tests won't actually run in CI until the lint backlog is addressed or the dependency is loosened. Flagging for visibility.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

https://claude.ai/code/session_01VB5AzgeVSgdKfAGDbrUVh6

### CODE REVIEW

File: tests/unit/test_security_enforcer.py [hunk 1]
```diff
@@ -0,0 +1,374 @@
+"""Tests for SecurityEnforcerMiddleware.
+
+Verifies input sanitization, content-type enforcement, auth verification,
+HTTPS enforcement, audit logging, and response headers.
+"""
+
+from __future__ import annotations
+
+import json
```
[the remark this answers is not in the document]
Comment (irfankabir02) [reply tests/unit/test_security_enforcer.py]: @claude
